File: 1189.maximum-number-of-balloons.cpp
/*
 * @lc app=leetcode id=1189 lang=cpp
 *
 * [1189] Maximum Number of Balloons
 */
#include <vector>
#include <string>
#include <algorithm>
using namespace std;
// @lc code=start
class Solution
{
public:
    int maxNumberOfBalloons(string text)
    {
        vector<int> count(26);
        for (auto c : text)
        {
            ++count[c - 'a'];
        }
        return min({count['b' - 'a'], count['a' - 'a'], count['l' - 'a'] / 2, count['o' - 'a'] / 2, count['n' - 'a']});
    }
};
// @lc code=end

File: java/1189.maximum-number-of-balloons.java
/*
 * @lc app=leetcode id=1189 lang=java
 *
 * [1189] Maximum Number of Balloons
 */

// @lc code=start
class Solution {
    public int maxNumberOfBalloons(String text) {
        int[] count = new int[26];
        int len = text.length();
        for (int i = 0; i < len; ++i) {
            ++count[text.charAt(i) - 'a'];
        }
        return Math.min(count['b' - 'a'], Math.min(count['a' - 'a'],
                Math.min(count['l' - 'a'] / 2, Math.min(count['o' - 'a'] / 2, count['n' - 'a']))));
    }
}
// @lc code=end
